impress-org/givewp
# Feature: add convertkit add-on to the form migration process

Resolves # [GIVE-455]

## Description

This PR adds a step in the form migration process to convert ConvertKit data from v2 forms to v3 forms and includes the methods required to facilitate this in the FormMetaDecorator. 

It also includes unit tests on the form migration process for the Convertkit add-on. Three tests were added that check when migrating a form that uses the global settings, migrating a form that has per-form settings & migrating a form without meta.


## Additional context
While Migrating the form if enabled globally from the global settings is selected the form migration should assume that it should migrate the form with the add-on block. On the v2 form if "use default" is selected the form should migrate with global settings. If "customize" is selected it should migrate via the form settings. If "disabled" is selected while enabled globally is selected it should not migrate the form with the add-on block. 

## Affects
Form Migration

## Visuals

https://github.com/impress-org/givewp/assets/75056371/8460af7c-8b9f-449e-9593-3dbe307e9bf6


https://github.com/impress-org/givewp/assets/75056371/47605dce-557f-433d-b2ff-1b4c2742fa95


## Testing Instructions
- Set your global settings
- add a v2 form -> then migrate the form without editing the form settings.
- Verify the v3 form has the correct settings set from the global options.
- Create another v2 form -> update the per-form settings on the right -> then migrate the form
- Verify the v3 form has the correct settings set from the form options.

## Pre-review Checklist




[GIVE-455]: https://stellarwp.atlassian.net/browse/GIVE-455?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
